### PR TITLE
X-wso2-basepath support for Api definition

### LINKED ISSUE #1241: X-wso2-basepath support for Envoy gateway
### Describe your problem(s)

Need to implement the x-wso2-basepath extension support for envoy gateway.
### Describe your solution

Currently we  expose the api with default basepath.
default basepath: /api/v3/
curl  **http://localhost:10000/api/v3/pet/5** -v

Need to support a x-wso2 basepath and after that user can expose the Api like below
x-wso2-basepath: /petstore/v2
curl **http://localhost:10000/petstore/v2/pet/5** -v

### How will you implement it

---

### PR DESCRIPTION
**Pull request by chashikajw**
### Purpose

As $subject this will make the support of  x-wso2-basepath vendor extension. 

As an example if x-wso2-basepath extension is included in the swagger, it will replace the default basepath of the Api.

default basepath: /api/v3
x-wso2-basepath: /petstore/v2
resource: /pet/{petId}

**if x-wso2-basepath exist,**
 can be exposed using   **http://localhost:10000/petstore/v2/pet/5**

**if x-wso2-basepath doesn't exist,**
can be exposed using  **http://localhost:10000/api/v3/pet/5**

### Issues
 Fixes #1241

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: internal/pkg/oasparser/models/apiDefinition/openApi.go [hunk 1]
```diff
@@ -126,3 +126,5 @@ func IsServerUrlIsAvailable(swagger3 openapi3.Swagger) bool {
 	}
 }
 
+
+
```
Comment (menakaj) [inline internal/pkg/oasparser/models/apiDefinition/openApi.go:130]: Remove these empty lines

File: internal/pkg/oasparser/config/config.go [hunk 2]
```diff
@@ -23,5 +23,5 @@ const (
 	LISTENER_PORT     uint32 = 10000
 	API_DEFAULT_PORT   uint32 = 8080
 	MANAGER_STATPREFIX    string = "ingress_http"
-	CLUSTER_CONNECT_TIMEOUT    = 5 * time.Second
+	CLUSTER_CONNECT_TIMEOUT    = 20 * time.Second
```
Comment (menakaj) [inline internal/pkg/oasparser/config/config.go:26]: These should be taken from a configuration file.

File: internal/pkg/oasparser/configGenerator.go [hunk 3]
```diff
@@ -51,20 +51,28 @@ func GetProductionSources(location string) ([]types.Resource, []types.Resource,
...
-	routeConfigNameP := "routeProd_" + strings.Replace(mgwSwaggers[0].GetTitle(), " ", "", -1) + mgwSwaggers[0].GetVersion()
+		vHostP, _ := e.CreateVirtualHost(vHost_NameP, routesP)
 
-	listnerProd := e.CreateListener(listenerNameP, routeConfigNameP, vHostP)
+		listenerNameP := "listenerProd_1"
+		routeConfigNameP := "routeProd_" + strings.Replace(mgwSwaggers[0].GetTitle(), " ", "", -1) + mgwSwaggers[0].GetVersion()
+
+		listnerProd := e.CreateListener(listenerNameP, routeConfigNameP, vHostP)
+
+
```
Comment (menakaj) [inline internal/pkg/oasparser/configGenerator.go:66]: Remove unnecessary lines. Fix in other places as well.

File: internal/pkg/oasparser/config/config.go (hunk 2 above)
Comment (chashikajw) [reply internal/pkg/oasparser/config/config.go:26]: that menas config.toml ?  so everything should be moved to config.toml ?

Comment (menakaj) [reply internal/pkg/oasparser/config/config.go:26]: Need to move the ones the user should configure...

Comment (chashikajw) [reply internal/pkg/oasparser/config/config.go]: done

File: internal/pkg/oasparser/configGenerator.go (hunk 3 above)
Comment (chashikajw) [reply internal/pkg/oasparser/configGenerator.go:61]: done

**Comment by chashikajw** [comment]
> Add new line

**Comment by chashikajw** [comment]
> Add new line at the end of the file

done

File: config/tomlConfigParser.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package config
```
Comment (pubudu538) [inline config/tomlConfigParser.go:1]: Add the license header.

Comment (chashikajw) [reply config/tomlConfigParser.go:1]: This is already fixed

**Pull request merged**